### PR TITLE
[224] [Scaffold] Balance Breakdown

### LINKED ISSUE #224: [Scaffold] Balance Breakdown

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #224 

- full screen cover for the balance breakdown
- clearing out the background so we can do semi transparency
- bindings for the full screen cover
- latest block business logic
- mocked auto shielding threshold
- unit tests
- snapshot tests

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [x] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [x] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: secant/Wrappers/WrappedSDKSynchronizer.swift [hunk 1]
```diff
@@ -184,7 +188,19 @@ class LiveWrappedSDKSynchronizer: WrappedSDKSynchronizer {
         
         return .none
     }
-    
+
+    func getTransparentBalance() -> Effect<WalletBalance, Never> {
```
Comment (pacu) [inline secant/Wrappers/WrappedSDKSynchronizer.swift:192]: see above

**Review by pacu — CHANGES_REQUESTED** [review]

File: secant.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -203,8 +203,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/zcash/ZcashLightClientKit",
       "state" : {
-        "branch" : "master",
-        "revision" : "fba4cecbe61cce424ada9fe1f98b05b88d5c8920"
+        "revision" : "a2283f0171cffb79c423e51fdfec4a5e2fd4959b",
+        "version" : "0.16.5-beta"
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secant.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:207]: updated

File: secant/Dependencies/ZCashSDKEnvironment.swift [hunk 3]
```diff
@@ -33,29 +33,29 @@ struct ZCashSDKEnvironment {
 extension ZCashSDKEnvironment {
     static let mainnet = ZCashSDKEnvironment(
         defaultBirthday: BlockHeight(ZcashSDKConstants.defaultBlockHeight),
-        endpoint: LightWalletEndpoint(address: ZcashSDKConstants.endpointMainnetAddress, port: ZcashSDKConstants.endpointPort),
+        endpoint: LightWalletEndpoint(address: ZcashSDKConstants.endpointMainnetAddress, port: ZcashSDKConstants.endpointPort, secure: true),
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secant/Dependencies/ZCashSDKEnvironment.swift:36]: updated

File: secant/Features/BalanceBreakdown/BalanceBreakdownStore.swift [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            return Effect(value: .updateSynchronizerStatus)
+            
+        case .synchronizerStateChanged(let synchronizerState):
+            return Effect(value: .updateSynchronizerStatus)
+            
+        case .updateSynchronizerStatus:
+            return .merge(
+                Effect(value: .updateLatestBlock),
+                
+                environment.SDKSynchronizer.getShieldedBalance()
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secant/Features/BalanceBreakdown/BalanceBreakdownStore.swift:89]: updated

File: secant/Features/BalanceBreakdown/BalanceBreakdownStore.swift [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+extension BalanceBreakdownEnvironment {
+    static let live = BalanceBreakdownEnvironment(
+        numberFormatter: .live(),
+        SDKSynchronizer: LiveWrappedSDKSynchronizer(),
+        scheduler: DispatchQueue.main.eraseToAnyScheduler()
+    )
+
+    static let mock = BalanceBreakdownEnvironment(
+        numberFormatter: .live(),
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secant/Features/BalanceBreakdown/BalanceBreakdownStore.swift:57]: solved

File: secant/Utils/ClearBackgroundView.swift [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+//  Created by Lukáš Korba on 04.08.2022.
+//
+
+import SwiftUI
+
+struct ClearBackgroundView: UIViewRepresentable {
+    func makeUIView(context: Context) -> UIView {
+        let view = UIView()
+        DispatchQueue.main.async {
+            view.superview?.superview?.backgroundColor = .clear
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secant/Utils/ClearBackgroundView.swift:14]: It mas seem odd and I was surprised as well when I got inspired by some stack overflow thread how to solve it but without it the SwiftUI modifier for the background doesn't work. It doesn't crash but it's not a transparent view.

File: secant/Wrappers/WrappedSDKSynchronizer.swift [hunk 7]
```diff
@@ -83,6 +85,7 @@ class LiveWrappedSDKSynchronizer: WrappedSDKSynchronizer {
     private(set) var stateChanged: CurrentValueSubject<WrappedSDKSynchronizerState, Never>
     private(set) var notificationCenter: WrappedNotificationCenter
     private(set) var walletBirthday: BlockHeight?
+    private(set) var latestScannedHeight: BlockHeight?
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secant/Wrappers/WrappedSDKSynchronizer.swift:88]: refactored

File: secant/Wrappers/WrappedSDKSynchronizer.swift (hunk 1 above)
Comment (LukasKorba) [reply secant/Wrappers/WrappedSDKSynchronizer.swift:192]: yes, with 0.16.5-beta it simplified a lot of things, I refactored a lot of code. Also unit tests

File: secant/Utils/ClearBackgroundView.swift (hunk 6 above)
[the remark this answers is not in the document]
Comment (pacu) [reply secant/Utils/ClearBackgroundView.swift:14]: uhm... ok. Let's document why we use this and where the workaround come from and what problem it solves so we don't forget.

**Review by pacu — APPROVED** [review]

**Comment by pacu** [comment]
LGTM! Let's document that clear view workaround and it's done!

File: secant/Utils/ClearBackgroundView.swift (hunk 6 above)
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secant/Utils/ClearBackgroundView.swift:19]: documented + link to the proposed solution on stackoverflow added

**Review by pacu — APPROVED** [review]

**Pull request merged**